lib/spy/subroutine.rb

```diff
@@ -364,9 +365,14 @@ def get_spies(base_object, singleton_methods = true)
 
       # @private
       def get_spy_id(method)
-        return nil unless method.parameters[0].is_a?(Array)
-        id = method.parameters[0][1].to_s.sub!("__spy_args_", "")
-        id.to_i if id
+        return nil unless method.parameters[0].is_a?(Array) &&
```

Comment:
Awesome work! How do you feel about this instead to make it a bit more readable?

```ruby
if method.parameters[0].is_a?(Array)
  raw_id = method.parameters[0][1]
  if raw_id.to_s.start_with?(SPY_ARGS_PREFIX)
    slice_length = raw_id.length - SPY_ARGS_PREFIX.length
    raw_id[SPY_ARGS_PREFIX.length..-1].to_i
  end
end
```

If we want to prevent the range creation every time we could always move it into a constant also. I feel that is a micro optimization and wouldn't help that much.